lib/domain/qr_code/provider/scanner_provider.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:superformula_test/core/service_locator.dart';
+import 'package:superformula_test/domain/qr_code/model/validator_model.dart';
+import 'package:superformula_test/domain/qr_code/repository/qr_code_repo.dart';
+
+class ScannerProvider extends ChangeNotifier {
+  final RestClient client = ServiceLocator.locator<RestClient>();
+
+  ScannerProvider();
+
+  String? result = '';
```

Comment:
question: do these need to be initialized?

Reply:
Seeing that I had used the value in the `QrScannerPage`, and I had not checked for null at that point before the toUpper, initializing it had allowed for it not to be null.
Code in reference
`'${provider.format!.toUpperCase()}: ${provider.result!.toUpperCase()}',`